src/pages/iou/NewDistanceRequestPage.tsx

```diff
@@ -0,0 +1,61 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback, useEffect} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import DistanceRequest from '@components/DistanceRequest';
+import Navigation from '@libs/Navigation/Navigation';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
+import * as IOU from '@userActions/IOU';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {IOU as IOUType, Report} from '@src/types/onyx';
+
+type NewDistanceRequestPageOnyxProps = {
+    iou: OnyxEntry<IOUType>;
+    report: OnyxEntry<Report>;
+};
+
+type NewDistanceRequestPageProps = NewDistanceRequestPageOnyxProps & StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.DISTANCE>;
+
+// This component is responsible for getting the transactionID from the IOU key, or creating the transaction if it doesn't exist yet, and then passing the transactionID.
+// You can't use Onyx props in the withOnyx mapping, so we need to set up and access the transactionID here, and then pass it down so that DistanceRequest can subscribe to the transaction.
+function NewDistanceRequestPage({iou, report, route}: NewDistanceRequestPageProps) {
+    const iouType = route.params.iouType ?? 'request';
```

Comment:
```suggestion
    const iouType = route.params.iouType;
```

Theoretically this is unnecessary since `route.params.iouType` is always defined.